### PR TITLE
Add travis file

### PR DESCRIPTION
**Pull request by lucyleeow**

### CODE REVIEW

**Comment by lucyleeow** [comment]
Hmm there are some flake8 problems. Should I fix here?

**Comment by lucyleeow** [comment]
@maikia 
> ./plot_topomap.py:12:10: F821 undefined name 'X'

**Comment by lucyleeow** [comment]
Excluding `plot_topomap.py` as suggested by @maikia as this should only be used within notebook.

**Comment by lucyleeow** [comment]
It may be worthwhile to offer a quick version of `ramp-test`. Travis is failing due to time out:

> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

it may be nice for participants to be able to quickly test their code too.

We can train and test on a subset of the data. Not sure what is best, restrict number of subjects and samples?

File: problem.py [hunk 1]
```diff
@@ -170,9 +170,19 @@ def get_cv(X, y):
 
 def _read_data(path, dir_name):
     X_df = pd.read_csv(os.path.join(DATA_HOME, dir_name, 'X.csv.gz'))
-    y_sparse = sparse.load_npz(
+    y = sparse.load_npz(
         os.path.join(DATA_HOME, dir_name, 'target.npz')).toarray()
-    return X_df, y_sparse
+    test = os.getenv('RAMP_TEST_MODE', 0)
+    if test:
```
Comment (maikia) [inline problem.py:176]: Cool. Thanks. Maybe add a small note beforehand saying that we are from now on working on limited data and that they should use the full data for their submissions

Comment (lucyleeow) [reply problem.py:176]: This is only case when you do `ramp-test --quick-test`

Comment (maikia) [reply problem.py:186]: Then maybe it's good to also inform them that they can also use this command?

**Comment by lucyleeow** [comment]
Problem with scoring i think
```
  File "/home/travis/miniconda/envs/testenv/lib/python3.8/site-packages/rampwf/utils/submission.py", line 371, in bag_submissions

    pred, scores = get_score_cv_bags(

  File "/home/travis/miniconda/envs/testenv/lib/python3.8/site-packages/rampwf/utils/combine.py", line 65, in get_score_cv_bags

    score_cv_bags.append(score_type.score_function(

  File "/home/travis/miniconda/envs/testenv/lib/python3.8/site-packages/rampwf/score_types/base.py", line 21, in score_function

    return self.__call__(y_true, y_pred)

  File "./problem.py", line 35, in __call__

    score = hamming_loss(y_true_proba, y_proba)

  File "/home/travis/miniconda/envs/testenv/lib/python3.8/site-packages/sklearn/utils/validation.py", line 73, in inner_f

    return f(**kwargs)

  File "/home/travis/miniconda/envs/testenv/lib/python3.8/site-packages/sklearn/metrics/_classification.py", line 2098, in hamming_loss

    y_type, y_true, y_pred = _check_targets(y_true, y_pred)

  File "/home/travis/miniconda/envs/testenv/lib/python3.8/site-packages/sklearn/metrics/_classification.py", line 83, in _check_targets

    type_pred = type_of_target(y_pred)

  File "/home/travis/miniconda/envs/testenv/lib/python3.8/site-packages/sklearn/utils/multiclass.py", line 287, in type_of_target

    _assert_all_finite(y)

  File "/home/travis/miniconda/envs/testenv/lib/python3.8/site-packages/sklearn/utils/validation.py", line 97, in _assert_all_finite

    raise ValueError(

ValueError: Input contains NaN, infinity or a value too large for dtype('float64').
```

**Comment by agramfort** [comment]
can you replicate the pb https://travis-ci.com/github/ramp-kits/meg/builds/177058895 locally?

**Comment by lucyleeow** [comment]
> can you replicate the pb https://travis-ci.com/github/ramp-kits/meg/builds/177058895 locally?

Yes

**Comment by agramfort** [comment]
I cannot replicate. It works on my machine...

**Comment by lucyleeow** [comment]
Weird!

I only tried `ramp-test` today and I always got this error but @maikia said it was working fine for her before. Is it possible that the data files changed? Did you download the data files again with `download_data.py`?

**Comment by agramfort** [comment]
yes I redownloaded the data

>

**Comment by lucyleeow** [comment]
@agramfort I tried again with a fresh environment, used `pip install -r requirements.txt`. Still get the error.

@maikia was also getting the error. Also `ramp-test` (without `quick-test`) is extremely slow (>10min), where it was much faster for her before.

**Comment by agramfort** [comment]
I tried in a fresh conda env using python 3.7 and it works for me:

conda create --name minimal python=3.7
conda activate minimal
pip install -r requirements.txt
ramp-test --quick-test
ramp-test --submission lasso_lars --quick-test

wdyt?

>

**Comment by lucyleeow** [comment]
Hmm my Python version: Python 3.8.3

Could this be problem - does `dtype('float64')` change between Python versions?

**Comment by agramfort** [comment]
ok I manage to replicate using 3.8 !

let me investigate

**Comment by lucyleeow** [comment]
Thanks. FYI travis is also using 3.8

**Comment by lucyleeow** [comment]
When we combine predictions, it is only defined for none NaN values:
```Python
        combined_predictions.y_pred[np.less(
            combined_predictions.y_pred,
            0.5,
            where=~np.isnan(combined_predictions.y_pred)
            )] = 0.0
        combined_predictions.y_pred[np.greater_equal(
            combined_predictions.y_pred,
            0.5,
            where=~np.isnan(combined_predictions.y_pred)
            )] = 0.0
```

**Comment by agramfort** [comment]
GRRR !!! I cannot replicate if I rerun the code twice. it's an eisenbug !
I have rerun

ramp-test --submission lasso_lars --quick-test

10 times and now never crashes... :( :(

>

**Comment by maikia** [comment]
for the record:
1. All was working fine for me previously: quick, no error. 
2. After recent changes I didn't test it using `ramp-test` until today.
3. In last few days I made software updates on my comp and since then my RAMP start failing (when deploying locally), I had to install some additional missing library:
```pip install --upgrade GitPython==2.1.15 pip install --upgrade gitdb2==2.0.6 gitdb==0.6.4```
, it worked fine afterwards.
4. Today, at first, after I run `ramp-test` the first error was not bagging but joblib:

```
FileNotFoundError: [Errno 2] No such file or directory: '/dev/shm/joblib_memmapping_folder_5546_4224768678/5546-140337661454992-d01463f445f84999808198aaf7ac0057.pkl'
'''
joblib.externals.loky.process_executor.BrokenProcessPool: A task has failed to un-serialize. Please ensure that the arguments of the function are all picklable.
```
I updated joblib. Since then `ramp-test` (with or w/out --quick-test) is extremely slow and gives the 

```ValueError: Input contains NaN, infinity or a value too large for dtype('float64').``` in the end

**Comment by lucyleeow** [comment]
Could it be an OS problem. I consistently get the error, I think @maikia does too. Travis is also linux.

This is my minimal environment:
```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
ca-certificates           2020.6.24                     0  
certifi                   2020.6.20                py38_0  
click                     7.1.2                    pypi_0    pypi
cloudpickle               1.5.0                    pypi_0    pypi
cycler                    0.10.0                   pypi_0    pypi
joblib                    0.16.0                   pypi_0    pypi
kiwisolver                1.2.0                    pypi_0    pypi
ld_impl_linux-64          2.33.1               h53a641e_7  
libedit                   3.1.20191231         h14c3975_1  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.1.0                hdf63c60_0  
libstdcxx-ng              9.1.0                hdf63c60_0  
matplotlib                3.3.0                    pypi_0    pypi
ncurses                   6.2                  he6710b0_1  
numpy                     1.19.1                   pypi_0    pypi
openssl                   1.1.1g               h7b6447c_0  
pandas                    1.0.5                    pypi_0    pypi
pillow                    7.2.0                    pypi_0    pypi
pip                       20.1.1                   py38_1  
pyparsing                 2.4.7                    pypi_0    pypi
python                    3.8.3                hcff3b4d_2  
python-dateutil           2.8.1                    pypi_0    pypi
pytz                      2020.1                   pypi_0    pypi
pyyaml                    5.3.1                    pypi_0    pypi
ramp-utils                0.6.1                    pypi_0    pypi
ramp-workflow             0.3.3                    pypi_0    pypi
readline                  8.0                  h7b6447c_0  
scikit-learn              0.23.1                   pypi_0    pypi
scipy                     1.5.2                    pypi_0    pypi
setuptools                49.2.0                   py38_0  
six                       1.15.0                   pypi_0    pypi
sqlite                    3.32.3               h62c20be_0  
threadpoolctl             2.1.0                    pypi_0    pypi
tk                        8.6.10               hbc83047_0  
wheel                     0.34.2                   py38_0  
xz                        5.2.5                h7b6447c_0  
zlib                      1.2.11               h7b6447c_3  
```

Also added Python 3.6 and 3.7 to travis.

**Comment by lucyleeow** [comment]
All Python versions have the same error on travis. Potentially due to OS ?

**Comment by agramfort** [comment]
I could investigate on dragos. I suspect a bug in the `np.less(..., where=...)`. Just removing it fixes it and does not generate any warning.

you owe me a 🍺 @lucyleeow and @maikia

**Review by agramfort — APPROVED** [review]

**Comment by lucyleeow** [comment]
>  I suspect a bug in the `np.less(..., where=...)`

To be fair I did point this out :stuck_out_tongue:

**Comment by agramfort** [comment]
Let’s see this as a team success. We all deserve a beer 🍺;)

Feel free to merge

**Comment by maikia** [comment]
Thanks @agramfort and @lucyleeow

**Pull request merged**